experiments/tree_tensor_network/ttn_1d_uniform.py

```diff
@@ -932,11 +1966,180 @@ def get_ham_ising_tube(dtype, Ly, lam=-3.044):
       for i in range(Ly)
   ]
 
-  Xcol = [tf.convert_to_tensor(Xc, dtype=dtype) for Xc in Xcol]
-  Zcol = [tf.convert_to_tensor(Zc, dtype=dtype) for Zc in Zcol]
+  Xcol = [convert_to_tensor(Xc, dtype=dtype) for Xc in Xcol]
+  Zcol = [convert_to_tensor(Zc, dtype=dtype) for Zc in Zcol]
 
   h1 = lam * sum(Zcol) - sum(Xcol[i] @ Xcol[(i + 1) % Ly] for i in range(Ly))
 
   h_mpo_2site = ([-Xc for Xc in Xcol], Xcol)
 
   return h1, h_mpo_2site
+
+
+def set_backend(backend):
+  """Sets the backend to use for tree tensor network computations.
+
+  A backend must be set after importing the module.
+
+  Args:
+    backend: Possible values are "tensorflow", "jax", and "numpy".
+  """
+  tensornetwork.set_default_backend(backend)
+
+  global np
```

Comment:
Wow. This file terrifies me. Why are you globaling everything?

Reply:
This is a really really bad way to swap backends, and it definitely going to break in the future. It's way too easy to miss an assignment that will cause the code to crash, or even worse fail silently. Why didn't you go with the object oriented approach we used eariler?

Reply:
I feel like you are unnecessarily terrified by globals! 😄
Anyway, yes, I warned you the backend stuff was ugly. I'm not sure I see why it's more likely to crash or fail silently (why would it fail silently?) than the OO approach, but the OO approach definitely produces prettier backend code. The main reason I didn't do it was basically an inclination to avoid plastering the rest of the file with "backend" arguments (the tree code is not really OO at all, which makes it easier to use tf.defun/tf.function with it).

Reply:
Would you be happy with a global backend object?

Reply:
You could do what we do now with the `set_default_backend` and make a module object.

And the reason I'm worried is that we both have a `switch_backend` method and a new guy who is trying to bring in support for Pytorch right now. Like I said, it's too easy to miss something with the current implementation (vs always throwing a `NotImplementedError` in the `BaseBackend` object).

Reply:
I realized a reason you might be more concerned... a missing definition might mean one is left with a function from the wrong backend after switching more than once. That *would* be nasty...